Adding backward compatibility for sharding support in old DataLoader

VitalyFedyunin opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #61312 Sort imports of test_datapipe.py
* **#61237 Adding backward compatibility for sharding support in old DataLoader**
* #61236 Implement usage of `is_shardable` and `apply_sharding`
* #61235 Add DataPipes Graph Functions
* #61234 [DataLoader] Adding demux and mux DataPipe-s

Differential Revision: [D29588832](https://our.internmc.facebook.com/intern/diff/D29588832)

VitalyFedyunin committed:
Adding backward compatibility for sharding support in old DataLoader

[ghstack-poisoned]

VitalyFedyunin committed:
Update on "Adding backward compatibility for sharding support in old DataLoader"




[ghstack-poisoned]

VitalyFedyunin committed:
Update on "Adding backward compatibility for sharding support in old DataLoader"




[ghstack-poisoned]

VitalyFedyunin committed:
Update on "Adding backward compatibility for sharding support in old DataLoader"




[ghstack-poisoned]

VitalyFedyunin committed:
Update on "Adding backward compatibility for sharding support in old DataLoader"




[ghstack-poisoned]

VitalyFedyunin committed:
Update on "Adding backward compatibility for sharding support in old DataLoader"




[ghstack-poisoned]

VitalyFedyunin committed:
Update on "Adding backward compatibility for sharding support in old DataLoader"




[ghstack-poisoned]

ejguan commented on test/test_datapipe.py:
```diff
@@ -1288,29 +1290,42 @@ def test_traverse_forked(self):
 
 class TestSharding(TestCase):
+
+    def get_pipeline(self):
```
nit: Add an underscore in front of this function can prevent unittest over itself.

ejguan reviewed (approved):
LGTM.

VitalyFedyunin committed:
Update on "Adding backward compatibility for sharding support in old DataLoader"




[ghstack-poisoned]

VitalyFedyunin committed:
Update on "Adding backward compatibility for sharding support in old DataLoader"




[ghstack-poisoned]

VitalyFedyunin commented:
@VitalyFedyunin has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29588832).

VitalyFedyunin committed:
Update on "Adding backward compatibility for sharding support in old DataLoader"


Differential Revision: [D29588832](https://our.internmc.facebook.com/intern/diff/D29588832)

[ghstack-poisoned]

VitalyFedyunin commented:
@VitalyFedyunin has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29588832).
